src/main/java/com/google/devtools/build/lib/server/GrpcServerImpl.java

```diff
@@ -512,7 +512,9 @@ private void executeCommand(RunRequest request, BlockingStreamObserver<RunRespon
           option.getOption().toString(StandardCharsets.ISO_8859_1)));
     }
 
-    try (RunningCommand command = commandManager.create()) {
+    commandManager.preemptEligibleCommands();
+
+    try (RunningCommand command = request.getPreemptible() ? commandManager.createPreemptibleCommand() : commandManager.createCommand()) {
```

Comment:
ok, i guess i can see how the two methods thing i mentioned can be awkward, but i think it's still a net win (no action required here btw)

Reply:
I agree. It's definitely more explicit, even at the cost of a bit of duplicated code. These sort of scenarios make me appreciate keyword args in other languages.